From df4c567c35c317b21dbc957f72c225840238363b Mon Sep 17 00:00:00 2001
From: Preston Peranich <pperanich@gmail.com>
Date: Mon, 19 Feb 2024 16:46:39 -0500
Subject: [PATCH] Expand env vars and tilde for search_dirs

---
 src/main.rs | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/src/main.rs b/src/main.rs
index 381b680..d17c1be 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -42,7 +42,23 @@ fn main() -> Result<(), TmsError> {
         SubCommandGiven::No(config) => config, // continue
     };
 
-    let mut search_dirs = config.search_dirs.unwrap_or(Vec::new());
+    let mut search_dirs: Vec<_> = config
+        .search_dirs
+        .unwrap_or(Vec::new())
+        .into_iter()
+        .map(|mut search_dir| {
+            let expanded_path = shellexpand::full(&search_dir.path.to_string_lossy())
+                .change_context(TmsError::IoError)
+                .unwrap()
+                .to_string();
+
+            search_dir.path = canonicalize(expanded_path)
+                .change_context(TmsError::IoError)
+                .unwrap();
+
+            search_dir
+        })
+        .collect();
 
     // merge old search paths with new search directories
     if let Some(search_paths) = config.search_paths {
-- 
2.43.0

